include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_scan.h

```diff
@@ -130,86 +116,63 @@ single_pass_scan_impl(sycl::queue __queue, _InRange&& __in_rng, _OutRange&& __ou
     static_assert(_Inclusive, "Single-pass scan only available for inclusive scan");
 
     const ::std::size_t n = __in_rng.size();
-    auto __max_cu = __queue.get_device().template get_info<sycl::info::device::max_compute_units>();
-    //std::size_t num_wgs = __max_cu;
-    //std::size_t num_wgs = 448;
-    //std::size_t num_wgs = 256;
-
-    // TODO: use wgsize and iters per item from _KernelParam
-    //constexpr ::std::size_t __elems_per_item = _KernelParam::data_per_workitem;
 #ifdef _ONEDPL_SCAN_ITER_SIZE
-    constexpr ::std::size_t __elems_per_item = _ONEDPL_SCAN_ITER_SIZE;
+    constexpr ::std::size_t __elems_per_workload = _ONEDPL_SCAN_ITER_SIZE;
 #else
-    constexpr ::std::size_t __elems_per_item = 8;
+    constexpr ::std::size_t __elems_per_workload = 8;
 #endif
     // Next power of 2 greater than or equal to __n
     auto __n_uniform = n;
     if ((__n_uniform & (__n_uniform - 1)) != 0)
         __n_uniform = oneapi::dpl::__internal::__dpl_bit_floor(n) << 1;
-    //std::size_t wgsize = n/num_wgs/__elems_per_item;
-    std::size_t wgsize = 256;
-    std::size_t num_items = __n_uniform/__elems_per_item;
-    std::size_t num_wgs = num_items/wgsize;
-    //
-    //std::size_t wgsize = 256;
-    //std::size_t num_items = 114688;
-
+    std::size_t num_workloads = __n_uniform / __elems_per_workload;
+    std::size_t wgsize = num_workloads > 256 ? 256 : num_workloads;
+    std::size_t num_wgs = oneapi::dpl::__internal::__dpl_ceiling_div(num_workloads, wgsize);
 
     constexpr int status_flag_padding = 32;
-    std::uint32_t status_flags_size = num_wgs+1+status_flag_padding;
-
-    //printf("launching kernel items=%lu wgs=%lu wgsize=%lu elems_per_iter=%lu max_cu=%u\n", num_items, num_wgs, wgsize, __elems_per_item, __max_cu);
+    std::uint32_t status_flags_size = num_wgs + status_flag_padding;
 
     uint32_t* status_flags = sycl::malloc_device<uint32_t>(status_flags_size, __queue);
-    //__queue.memset(status_flags, 0, status_flags_size * sizeof(uint32_t));
+    // First status_flags_size elements: partial sums of each workgroup
+    // First status_flags_size elements: full results / full sums, i.e. sum of the previous and current workgroup's partial sums
+    _Type* sums = sycl::malloc_device<_Type>(status_flags_size * 2, __queue);
```

Comment:
Yes. I am not keen of using `tile_sums` either, as we should consider for a generic BinaryOp. But Since the algorithm only works for sums for now, we can let it be for now.